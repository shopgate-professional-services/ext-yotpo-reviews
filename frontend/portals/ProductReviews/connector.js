import { connect } from 'react-redux';
import { getProduct } from '@shopgate/pwa-common-commerce/product/selectors/product';
import { historyPush } from '@shopgate/pwa-common/actions/router';
/**
 * Maps the terms and conditions is orderable State.
 * @param {Object} state The current application state.
 * @param {Object} props The component props.
 * @returns {Object} The extended component props.
 */
const mapStateToProps = (state, props) => ({
  product: getProduct(state, props),
});

/**
 * Connects the dispatch function to a callable function in the props.
 * @param {Function} dispatch The redux dispatch function.
 * @return {Object} The extended component props.
 */
const mapDispatchToProps = dispatch => ({
  navigate: pathname => dispatch(historyPush({ pathname })),
});

export default connect(mapStateToProps, mapDispatchToProps);
