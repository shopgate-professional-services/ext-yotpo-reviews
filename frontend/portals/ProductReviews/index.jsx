import React from 'react';
import PropTypes from 'prop-types';
import { RouteContext } from '@shopgate/pwa-common/context';
import { hex2bin } from '@shopgate/pwa-common/helpers/data';
import HtmlReferenceContainer from '../../components/HtmlReferenceContainer';
import getConfig from '../../helpers/getConfig';
import connect from './connector';

const config = getConfig();

const HtmlReferenceContainerContext = connect(({
  productId,
  product,
  openPageExtern,
}) => (
  <HtmlReferenceContainer
    config={config}
    productId={productId}
    product={product}
    openPageExtern={openPageExtern}
  />
));

HtmlReferenceContainerContext.propTypes = {
  config: PropTypes.shape().isRequired,
  productId: PropTypes.string.isRequired,
  openPageExtern: PropTypes.func,
  productUrl: PropTypes.string,
};

HtmlReferenceContainerContext.defaultProps = {
  openPageExtern: () => {},
  productUrl: null,
};

/**
 * Renders YotpoWidget component in product.reviews portal
 * @returns {JSX}
 */
const ProductReviews = () => {
  if (!config.yotpoAppKey) {
    return null;
  }
  return (
    <RouteContext.Consumer>
      {({ params }) => (
        <HtmlReferenceContainerContext
          productId={hex2bin(params.productId)}
          config={config}
        />
    )}
    </RouteContext.Consumer>
  );
};

export default connect(ProductReviews);
