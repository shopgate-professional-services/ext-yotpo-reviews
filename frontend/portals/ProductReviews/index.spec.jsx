import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import { basicProductState } from '@shopgate/pwa-common-commerce/product/mock';
import { RouteContext } from '@shopgate/pwa-common/context';
import ProductReviews from './index';

const mockedStore = configureStore();
jest.mock('@shopgate/pwa-common-commerce/product/selectors/product', () => ({
  getProduct: jest.fn().mockReturnValue({ productId: 'product_one' }),
}));
jest.mock('../../components/HtmlReferenceContainer', () => () => (<div>HtmlReferenceContainer</div>));

/**
 * Create Component
 * @return {ReactWrapper}
 */
const createComponent = () => mount(
  <RouteContext.Provider value={{ params: { product: 'product_one' } }}>
    <Provider store={mockedStore({ product: basicProductState })}>
      <ProductReviews productId="product_one" />
    </Provider>
  </RouteContext.Provider>,
  mockRenderOptions
);

describe('ProductReviews', () => {
  it('should render', () => {
    const component = createComponent();
    expect(component).toMatchSnapshot();
  });
});
