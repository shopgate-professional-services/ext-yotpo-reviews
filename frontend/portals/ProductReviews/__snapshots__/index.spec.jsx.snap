// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReviews should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ProductReviews)
    productId="product_one"
  >
    <ProductReviews
      openPageExtern={[Function]}
      product={
        Object {
          "id": "product_one",
        }
      }
      productId="product_one"
    >
      <Connect(Component)
        config={
          Object {
            "yotpoAppKey": "LHgQXH7Uj3KgqA8WkfUEfwsf9gMzKi3acoVoU8on",
          }
        }
        openPageExtern={[Function]}
        productId={false}
        productUrl={null}
      >
        <Component
          config={
            Object {
              "yotpoAppKey": "LHgQXH7Uj3KgqA8WkfUEfwsf9gMzKi3acoVoU8on",
            }
          }
          openPageExtern={[Function]}
          product={
            Object {
              "id": "product_one",
            }
          }
          productId={false}
          productUrl={null}
        >
          <Component
            config={
              Object {
                "yotpoAppKey": "LHgQXH7Uj3KgqA8WkfUEfwsf9gMzKi3acoVoU8on",
              }
            }
            openPageExtern={[Function]}
            product={
              Object {
                "id": "product_one",
              }
            }
            productId={false}
          >
            <div>
              HtmlReferenceContainer
            </div>
          </Component>
        </Component>
      </Connect(Component)>
    </ProductReviews>
  </Connect(ProductReviews)>
</Provider>
`;
